pyuvdata/uvdata.py

```diff
@@ -2508,11 +2508,46 @@ def get_feedpols(self):
         else:
             return list(set(''.join(self.get_pols())))
 
-    def antpair2ind(self, ant1, ant2):
+    def antpair2ind(self, ant1, ant2=None, ordered=True):
         """
-        Get blt indices for given (ordered) antenna pair.
-        """
-        return np.where((self.ant_1_array == ant1) & (self.ant_2_array == ant2))[0]
+        Given an antenna pair key, return indices along the baseline-time axis.
+        This will search for either the key as specified, or the key and its
+        conjugate.
+
+        Args:
+            ant1, ant2:
+                Either an antenna-pair key, or key expanded as arguments.
+                Example: antpair2ind( (10, 20) ) or antpair2ind(10, 20)
+            ordered : Boolean, if True, search for antpair as provided, else
+                search for it and it conjugate.
+
+        Returns:
+            inds: int-64 ndarray containing indices of the antpair along the
+                baseline-time axis.
+        """
+        # check for expanded antpair or key
+        if ant2 is None:
+            assert isinstance(ant1, tuple), \
```

Comment:
Just to add a comment on this latest version: `assert` statements are not as robust as checking conditions with `if` and then raising an error, because one can run python in such a way (with the `-O` flag) that `asssert`s are completely ignored: https://stackoverflow.com/a/2167639.

`assert` could be fine for doing basic self-consistency checks, but in this case we'd rather raise an error if the user doesn't provide the right arguments.

Reply:
Hmm.. but i would argue these are self-consistency checks, in that, if I get rid of these `assert` statements and the user knows exactly how to use this function,  then nothing changes. These are only to help the novice user (and occasional advanced-user mistake) by enacting "self-consistency" checks on the data types: i.e. trying to be defensive in my programming.

Reply:
another way of saying it: the function will still error if given malformed inputs, but it will give a slightly less informative error. however, core functionality is not taken away if these assert statements vanished

Reply:
I went off and read a bit about using assertions vs exceptions in python. The consensus largely seems to be that assertions are intended for the developers of the package (indicating that they should chang/fix things), while Exceptions, especially ValueError, are for runtime issues caused by bad inputs by users.

> Assertions are for programmers. Exceptions are for errors that appear on runtime.

Reply:
I agree that nothing changes if the assert statements go away, but my general philosophy is that `assert` should be used to guarantee internal consistency of functions. If you're working directly with user input (which is the case here), then I think it's much safer to raise exceptions rather than waiting for the function to fail.

For instance, if this code were optimized and the user called `uvd.antpair2ind(0)`, instead of your helpful error message, they get `TypeError: 'int' object has no attribute '__getitem__'`, which would be thoroughly cryptic. I think it's better to raise an error in that case.